packages/design-system/src/components/_OcTextInputPassword/_OcTextInputPassword.vue

```diff
@@ -0,0 +1,125 @@
+<template>
+  <div ref="inputPasswordWrapper" class="oc-text-input-password-wrapper">
+    <input v-bind="$attrs" :type="showPassword ? 'text' : 'password'" @input="onInput" />
+    <oc-button
+      class="oc-text-input-show-password-toggle oc-px-s oc-background-default"
+      appearance="raw"
+      size="small"
+      @click="showPassword = !showPassword"
+    >
+      <oc-icon size="small" :name="showPassword ? 'eye-off' : 'eye'" />
+    </oc-button>
+  </div>
+  <portal v-if="showPasswordPolicyInformation" to="app.design-system.password-policy">
+    <div class="oc-text-small oc-flex oc-flex-column">
+      <span v-text="$gettext('Please enter a password that meets the following criteria:')" />
+      <div v-for="(rule, index) in passwordPolicy" :key="index" class="oc-flex oc-flex-middle">
+        <oc-icon
+          size="small"
+          :name="isPolicyRuleFulfilled(rule) ? 'check' : 'close'"
+          :variation="isPolicyRuleFulfilled(rule) ? 'success' : 'danger'"
+        />
+        <span :class="getPolicyMessageClass(rule)" v-text="getPolicyRuleMessage(rule)"></span>
+      </div>
+    </div>
+  </portal>
+</template>
+
+<script lang="ts">
+import { computed, defineComponent, ref, unref, watch } from 'vue'
+import OcIcon from '../OcIcon/OcIcon.vue'
+import OcButton from '../OcButton/OcButton.vue'
+import { useGettext } from 'vue3-gettext'
+export default defineComponent({
+  name: 'OCTextInputPassword',
+  components: { OcButton, OcIcon },
+  status: 'ready',
+  release: '1.0.0',
+  inheritAttrs: true,
+  props: {
+    passwordPolicy: {
+      type: Array,
+      default: () => []
+    }
+  },
+  emits: ['passwordChallengeCompleted', 'passwordChallengeFailed'],
+  setup(props, { emit }) {
+    const { $gettext } = useGettext()
+    const showPassword = ref(false)
+    const passwordEntered = ref(false)
+    const password = ref('')
+    const showPasswordPolicyInformation = computed(() => {
+      return !!(props.passwordPolicy.length && unref(passwordEntered))
+    })
+
+    const onInput = (event) => {
+      passwordEntered.value = true
+      password.value = event.target.value
+    }
+
+    const getPolicyRuleMessage = (policyRule) => {
+      const explained = policyRule.explain()[0]
+      const paramObj = {}
+
+      for (var formatKey = 0; formatKey < explained.format.length; formatKey++) {
```

Comment:
Could you add a short comment explaining what you explained me yesterday?

```suggestion
      for (let formatKey = 0; formatKey < explained.format.length; formatKey++) {
```